elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/metric/BasicMetricFunction.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2019, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.datastores.aggregation.metadata.metric;
+
+import com.yahoo.elide.datastores.aggregation.metadata.models.FunctionArgument;
+import com.yahoo.elide.datastores.aggregation.metadata.models.MetricFunction;
+import com.yahoo.elide.request.Argument;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Basic implementation for Metric function.
+ */
+@EqualsAndHashCode(callSuper = true)
+@Data
+@AllArgsConstructor
+public abstract class BasicMetricFunction extends MetricFunction {
```

Comment:
Is there a reason to separate this class out from the MetricFunction class?

Reply:
The complicated functions might need to extend from `MetricFunction`.

Reply:
Let's discuss